src/Host/Client/Impl/Session/RSessionEvaluationCommands.cs

```diff
@@ -210,13 +210,18 @@ public static class RSessionEvaluationCommands {
             return evaluation.ExecuteAsync(script);
         }
 
-        public static Task SetCodePageAsync(this IRExpressionEvaluator evaluation, int codePage, CancellationToken cancellationToken = default(CancellationToken)) {
+        public static async Task SetCodePageAsync(this IRExpressionEvaluator evaluation, int codePage, CancellationToken cancellationToken = default(CancellationToken)) {
+            string cp = $".{codePage}";
             if (codePage == 0) {
-                // TODO: update this to get the codepage via platform agnostic service
-                codePage = 437;
+                var isWindows = await evaluation.EvaluateAsync<bool>(".Platform$OS.type == 'windows'", REvaluationKind.Normal, cancellationToken);
```

Comment:
I suspect we'll be doing this often enough in the future, it might be worth making it an extension method on `IRExpressionEvaluator`.